pandas/io/formats/style.py

```diff
@@ -178,6 +178,24 @@ def to_excel(self, excel_writer, sheet_name='Sheet1', na_rep='',
                         startcol=startcol, freeze_panes=freeze_panes,
                         engine=engine)
 
+    def to_html(self, buf=None, mode='w'):
+        """
+        see pull request #17545
+        Write pandas.io.formats.style.Styler to HTML file
+
+        Parameters
+        ----------
+        filename : string or ExcelWriter object
+            Output filename
+        mode : string, default 'w'
+            File i/o mode used as open(filename, mode)
+
+        """
+        if buf is None:  # print to stdout instead of writing to file
+            return self.render()
+        with open(buf, 'w') as fid:
```

Comment:
Can you add tests to ensure that this works for other file-like objects (e.g. `pathlib.Path`)? It will on python 3.6+, but also test on an older version (or just let travis catch it). You may find `pandas.io.common.get_filepath_or_buffer` and `pandas.io.common._get_handle` useful.

Reply:
Oh, and you've hardcoded `'w'` here for the mode. This should be `mode` instead.

Reply:
Thanks for catching the careless mistake